code/datums/abilities/flock/flockmind.dm

```diff
@@ -305,3 +305,23 @@
 		return 1
 	panel.Subscribe(user)
 
+////////////////////////////////
+
+/datum/targetable/flockmindAbility/createStructure
+	name = "Fabricate Structure"
+	desc = "Create a structure tealprint for your drones to construct onto."
+	icon_state = "fabstructure"
+	cooldown = 4
+	targeted = 0
+
+/datum/targetable/flockmindAbility/createStructure/cast()
+	var/resourcecost = null
+	var/structurewantedtype = null
+	var/structurewanted = input("Select which structure you would like to create", "Tealprint Selection", "cancel") as null|anything in list("Collector")
```

Comment:
my mistake but I should have probably suggested `null|text` instead of `null|anything` since you are working with text